### DISCUSSION TITLE
Expand Tenant with single Server to multi Server failed

### DISCUSSION THREAD

**Discussion by Clara12062** [minio/minio#19997] [Q&A]
On my test environment, I have a k8s node and a minio tenant deployed on the k8s cluster with the following configuration: one serverpool; pool-0: 1Server*2VolumesPerServer

After k8s cluster expands nodes, I want minio Tenant to expand as well. I add a new pool to the tenant: pool-1: 3server*2volumesperServer. But after the expansion, minio service failed to start, and the pod in the pool-0 gave the following error:
```text
Waiting for atleast 1 remote servers with valid configuration to be online
Following servers mismatch in their configuration [http://minio-pool-1-0.minio-hl.testminio.svc.cluster.local:9000/export0 has incorrect configuration: Expected endpoint http://minio-pool-0-0.minio-hl.testminio.svc.cluster.local/export0, seen http://minio-pool-0-0.minio-hl.testminio.svc.cluster.local:9000/export0 http://minio-pool-1-1.minio-hl.testminio.svc.cluster.local:9000/export0 has incorrect configuration: Expected endpoint http://minio-pool-0-0.minio-hl.testminio.svc.cluster.local/export0, seen http://minio-pool-0-0.minio-hl.testminio.svc.cluster.local:9000/export0 http://minio-pool-1-2.minio-hl.testminio.svc.cluster.local:9000/export0 has incorrect configuration: Expected endpoint http://minio-pool-0-0.minio-hl.testminio.svc.cluster.local/export0, seen http://minio-pool-0-0.minio-hl.testminio.svc.cluster.local:9000/export0]
Waiting for atleast 1 remote servers with valid configuration to be online
```
Why does this error occur? In the test environment, I tried the expansion of multi-server, and found that it can be normal expansion. It seems that only single node expansion will have problems, is this known?

**Comment by klauspost** [comment]
> http://minio-pool-1-0.minio-hl.testminio.svc.cluster.local:9000/export0 has incorrect configuration: Expected endpoint http://minio-pool-0-0.minio-hl.testminio.svc.cluster.local/export0, seen http://minio-pool-0-0.minio-hl.testminio.svc.cluster.local:9000/export0

Seems like you are missing a `:9000` somewhere. Your arguments must match exactly for both old and new servers.

**Reply by Clara12062** [reply]
yep~ But now it is done via the minio-operator, see /tmp/minio/config.env as follows:
```text
bash-4.4$ cat /tmp/minio/config.env 
export MINIO_BROWSER="on"
export MINIO_ROOT_USER="2ED4KKVDGHVY0UVT"
export MINIO_ROOT_PASSWORD="VCDFWQMMNNWNOGGVFHH5FPFFTPBQURC4"
export MINIO_ARGS="http://minio-pool-0-0.minio-hl.localminio.svc.cluster.local/export{0...1} http://minio-pool-1-{0...2}.minio-hl.localminio.svc.cluster.local/export{0...1}"
```
